libfwupdplugin/fu-kernel.c

```diff
@@ -305,3 +305,76 @@ fu_kernel_get_cmdline(GError **error)
 	return NULL;
 #endif
 }
+
+gboolean
+fu_kernel_is_cmdline_mutable(void)
+{
+	g_autofree gchar *grubby_path = fu_path_find_program("grubby", NULL);
+	return grubby_path != NULL;
+}
```

Comment:
As this information is going to be used not at the time you try to run the fix, but rather deciding whether you _CAN_ run the fix I think we also need two other checks:

1. Check whether the configuration file that stores kernel cmdline grubby stores to is writable.
2. Check whether the system is using a UKI to boot.  If it's using a UKI, how do you update kernel command line?

Reply:
There's another check we have in the fwupd codebase to check if a file is writable in `fu_config_add_location`.  I think you can generalize it into a `fu-common` function and use that, no?

Reply:
I added it inline, it was slightly different to the config case.